frontend/src/components/modules/Tournaments/OngoingTournament/RoundRobinTournamentView.tsx

```diff
@@ -12,64 +13,206 @@ import {
+    if (initialRender.current) {
+      initialRender.current = false;
+      updatePlayerStats();
+    }
+  }, []);
+
+  const getPlayerNames = (): void => {
+    setPlayers((prevPlayers) => {
+      const updatedPlayers = [...prevPlayers];
+      const playersObjects: User[] = tournament.players;
+      if (playersObjects.length > 0) {
+        for (const playerObject of playersObjects) {
+          const playerExists = updatedPlayers.some(
+            (player) => player.id === playerObject.id
+          );
+          if (!playerExists) {
+            updatedPlayers.push({
+              id: playerObject.id,
+              name: playerObject.firstName,
+              wins: 0,
+              losses: 0,
+              points: 0
+            });
+          }
+        }
       }
+      return updatedPlayers;
+    });
+  };
+
+  const sortMatches = (): void => {
+    const matches: Match[] = tournament.matchSchedule;
+    if (matches.length > 0) {
+      const sortedMatches = matches
+        .filter((match) => match.elapsedTime !== undefined)
+        .sort((a, b) => {
+          return b.elapsedTime - a.elapsedTime;
+        });
+      ongoingMatches = sortedMatches.filter(
+        (match) =>
+          match.elapsedTime !== null &&
+          match.elapsedTime > 0 &&
+          match.winner === undefined
+      );
+      upcomingMatches = sortedMatches.filter(
+        (match) =>
+          (match.elapsedTime === null || match.elapsedTime <= 0) &&
+          match.winner === undefined
+      );
+      pastMatches = sortedMatches.filter(
+        (match) =>
+          match.elapsedTime !== null &&
+          match.elapsedTime > 0 &&
+          match.winner !== undefined
+      );
+    }
+  };
+
+  const createMatchButtons = (): void => {
+    const createMatchButton = (
+      match: Match,
+      variant: "contained" | "contained" | "contained",
+      disabled?: boolean,
+      color?: "secondary"
+    ): JSX.Element => {
+      const player1 = players.find(
+        (player) => player.id === match.players[0].id
+      )?.name;
+      const player2 = players.find(
+        (player) => player.id === match.players[1].id
+      )?.name;
+
+      return (
+        <div style={{ marginBottom: "10px" }}>
+          <Button
+            variant={variant}
+            onClick={() => {
+              navigate(`/matches/${match.id}`);
+            }}
+            disabled={disabled}
+            color={color}
+          >
+            {`${player1} - ${player2}`}
+          </Button>
+        </div>
+      );
     };
 
-    void fetchMatchData();
-  }, []);
+    const ongoingElements = ongoingMatches.map((match, index) =>
+      createMatchButton(match, "contained")
+    );
+    const upcomingElements = upcomingMatches.map((match, index) =>
+      createMatchButton(match, "contained", true)
+    );
+    const pastElements = pastMatches.map((match, index) =>
+      createMatchButton(match, "contained", false, "secondary")
+    );
```

Comment:
I think that you could change the function signature of the `createMatchButton` to accept an object of type `ButtonProps`:

```tsx
const createMatchButton = (
  match: Match,
  props: ButtonProps
): JSX.Element => {
  ...
  return (
    <div style={{ marginBottom: "10px" }}>
      <Button
        onClick={() => {
          navigate(`/matches/${match.id}`);
        }}
        {...props}
      >
      ...
      </Button>
    </div>
  );
};
```

You could then pass any additional props like so:

```tsx
const pastElements = pastMatches.map((match, index) =>
  createMatchButton(match, {
    disabled: false,
    variant: "contained",
    color: "secondary"
  })
);
```